web/src/pages/insights/SummaryTable.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { ReactNode, useState } from 'react'
+import { Table as SUITable } from 'semantic-ui-react'
+import Table from '../../shared/components/Table'
+import { ThemeContext } from '../../shared/components/ThemeProvider'
+import HtmlTooltip from '../../shared/utilities/htmlTooltip'
+import { ProjectInsightsSummary } from '../../sm-api'
+import { ColumnKey, HeaderCell, summaryTableHeaderCellConfigs } from './HeaderCell'
+import { FooterCell, footerCellConfigs } from './SummaryTableFooterCell'
+
+interface SummaryTableProps {
+    filteredData: ProjectInsightsSummary[]
+    handleSelectionChange: (columnName: ColumnKey, selectedOptions: string[]) => void
+    getUniqueOptionsForColumn: (key: ColumnKey) => string[]
+    getSelectedOptionsForColumn: (key: ColumnKey) => string[]
+}
+
+interface PercentageCellProps {
+    percentage: number
+    tooltipContent: ReactNode
+    isDarkMode: boolean
+    getPercentageColor: (percentage: number, isDarkMode: boolean) => string
+}
+
+const PercentageCell: React.FC<PercentageCellProps> = ({
+    percentage,
+    tooltipContent,
+    isDarkMode,
+    getPercentageColor,
+}) => (
+    <SUITable.Cell
+        className="percentage-cell"
+        style={{ backgroundColor: getPercentageColor(percentage, isDarkMode) }}
+    >
+        <HtmlTooltip title={tooltipContent}>
+            <div>{percentage.toFixed(2)}%</div>
+        </HtmlTooltip>
+    </SUITable.Cell>
+)
+
+function calculatePercentage(count: number, total: number): number {
+    return count > 0 ? (count / total) * 100 : 0
+}
+
+function getPercentageColor(percentage: number, isDarkMode: boolean) {
+    const hue = (percentage / 100) * 120 // Convert percentage to hue value (0-120)
+    const saturation = isDarkMode ? '100%' : '90%' // Set saturation based on mode
+    const lightness = isDarkMode ? '25%' : '75%' // Set lightness based on mode
+
+    return `hsl(${hue}, ${saturation}, ${lightness})`
+}
+
+const getRowClassName = (sequencingType: string) => {
+    switch (sequencingType) {
+        case 'exome':
+            return 'exome-row'
+        case 'genome':
+            return 'genome-row'
+        default:
+            return 'rna-row'
+    }
+}
+
+const getSummaryCell = (
```

Comment:
This is a component, make sure to follow the standard component format + naming convention